src/ncp/ncp_base_dispatcher.cpp

```diff
@@ -0,0 +1,971 @@
+/*
+ *    Copyright (c) 2018, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+ *    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ *    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements general thread device required Spinel interface to the OpenThread stack.
+ */
+
+#include "ncp_base.hpp"
+
+namespace ot {
+namespace Ncp {
+
+// ----------------------------------------------------------------------------
+// MARK: Property Handler Jump Tables and Methods
+// ----------------------------------------------------------------------------
+NcpBase::PropertyHandler NcpBase::FindGetPropertyHandler(spinel_prop_key_t aKey)
+{
+    NcpBase::PropertyHandler handler;
+
+    switch (aKey)
+    {
+    case SPINEL_PROP_CAPS:
+        handler = &NcpBase::HandlePropertyGet<SPINEL_PROP_CAPS>;
+        break;
+    case SPINEL_PROP_DEBUG_TEST_ASSERT:
+        handler = &NcpBase::HandlePropertyGet<SPINEL_PROP_DEBUG_TEST_ASSERT>;
+        break;
+    case SPINEL_PROP_DEBUG_TEST_WATCHDOG:
+        handler = &NcpBase::HandlePropertyGet<SPINEL_PROP_DEBUG_TEST_WATCHDOG>;
+        break;
+    case SPINEL_PROP_DEBUG_NCP_LOG_LEVEL:
+        handler = &NcpBase::HandlePropertyGet<SPINEL_PROP_DEBUG_NCP_LOG_LEVEL>;
+        break;
```

Comment:
How about using a macro here to simply the list
```
#define NCP_GET_PROP_CASE(name)                                        \
    case SPINEL_PROP_##name:                                           \
        handler = &NcpBase::HandlePropertyGet<SPINEL_PROP_##_name>;    \
        break
```

and then...
```
    switch (aKey) 
    { 
    NCP_GET_PROP_CASE(CAPS);
    NCP_GET_PROP_CASE(TEST_ASSERT); 
    ....
    }

Reply:
This makes code shorter, but will this harm readability? Putting these function-like macros inside a switch looks a little bit strange in c/c++ syntax.

Reply:
I am fine with it as you have it originally.